docs/developer-docs/smart-contracts/best-practices/idempotency.mdx

```diff
@@ -15,8 +15,9 @@ In the case of network issues or other unexpected behavior, ICP clients (such as
 ingress update calls may be unable to determine whether their ingress
 request has been processed. For example, this can happen if the client loses its
 connection until the request status has been removed from the state tree, since ICP will remove the request from the system state tree some time after the ingress expiry.
+Similarly, canisters which call other canisters using calls with best-effort responses may be unable to determine whether the call was successful or not.
 
-This can be risky as the application might decide to retry transactions, potentially leading to serious security vulnerabilities such as double spending. 
+This can be risky as the callers (applications or canisters) might decide to retry transactions, potentially leading to serious security vulnerabilities such as double spending. 
```

Comment:
Dud you mean "users or canisters"? Or just something like a frontend running in a browser or a Web 2 app?

Reply:
Yes could also be users, I'll adjust